### ISSUE TITLE
Programmatically selecting date after scroll

### ISSUE THREAD

**Issue by ThasneemH** [patchthecode/JTAppleCalendar#866]
IMPORTANT - BEFORE CREATING YOUR ISSUE PLEASE READ THE FOLLOWING:

1. PLEASE STATE THE VERSION NUMBER you are using.
2. Are you using the latest version of JTAppleCalendar? Latest version is currently 7.1.5
3. PROVIDE ENOUGH INPORMATION SO I CAN RECREATE THE PROBLEM.

I have the latest version. 

I have a vertical scrolling month view calendar , below are 2 things I want to achieve:

1) I select a cell in the current visible month on the screen. Then to the next or previous month  and then scroll back again to the original month, the cell that was selected earlier still shows up selected. But I don't want that cell to be selected after the scroll. How do I achieve this? I tried  calling delselectAllDates : in  calendar(_ : didScrollToDateSegmentWith : ) , but that doesn't work. 

2) Another issue I have is that , once I scroll to every month , I want the first date to be selected. I can achieve this in calendar(_ : didScrollToDateSegmentWith : )  by calling programmaticallyselectItem  from within calendar(_ : didScrollToDateSegmentWith : ) . But since this delegate is called only after the scroll in completed, there is a small lag between when the section is visible on the screen versus when the date gets selected. Is there another  way to do this so that we  overcome this lag. thanks

**Comment by ThasneemH** [comment]
I see that  you have given the functionality in the patch below which might solve issue 2)  . I
will try it out.      https://github.com/patchthecode/JTAppleCalendar/issues/760

Will see if that resolved issue 1) as well. 
thanks

**Comment by patchthecode** [comment]
Do you have a sample application of github re-creating issue # 1 ?

**Comment by ThasneemH** [comment]
Thanks for the quick reply. I just found the cause of  issue 1) . Instead of using cellState.isSelected , I was using cell.isSelected , which was returning true even after calling deSelectAllDates. When I use cellState, it works fine now. I guess isSelected property of cell needs to be updated as well, so will leave this issue open for now. thanks.

**Comment by ThasneemH** [comment]
One question though, right now , using the cellState.isSelected instead of cell.isSelected, I have to call deSelectAllDates in calendar(_ : didScrollToDateSegmentWith : ) . Now when scrolling back to the section, i don't find the cell selected as expected.  But if I don't call deSelectAllDates, then when i scroll back , the cell that was originally selected is still selected - is this the expected behavior? that a selected cell stays selected ( regardless of scrolling) , unless we deselect it ( either by clicking on another cell or programmatically) The reason I ask is that , I went through the WLAppleCalendar example that does not call deSelect anywhere, but on scrolling after selection and then back the original section,  selected cell is not selected anymore.So just wanted to make sure. thanks

**Comment by patchthecode** [comment]
ah! i guess you are one of those devs that didnt read the big bold text on the main page :D .
`cellState.isSelected` is what we should use because cell.isSelected may be out of sync. It is explained in more detail on the main page.

**Comment by ThasneemH** [comment]
I see it now , I was careful with the willDisplayCell , but missed this one!  Thank u though.
